internal/pkg/mount/system.go

```diff
@@ -195,21 +199,32 @@ func SystemPartitionMount(r runtime.Runtime, label string, opts ...Option) (err
 		return err
 	}
 
+	mountpointsMutex.Lock()
+	defer mountpointsMutex.Unlock()
+
 	mountpoints[label] = mountpoint
 
 	return nil
 }
 
 // SystemPartitionUnmount unmounts a system partition by the label.
 func SystemPartitionUnmount(r runtime.Runtime, label string) (err error) {
-	if mountpoint, ok := mountpoints[label]; ok {
-		err = mountpoint.Unmount()
-		if err != nil {
-			return err
-		}
+	mountpointsMutex.RLock()
+	mountpoint, ok := mountpoints[label]
+	mountpointsMutex.RUnlock()
 
-		delete(mountpoints, label)
+	if !ok {
+		return nil
 	}
 
+	err = mountpoint.Unmount()
```

Comment:
Do we actually call Unmount concurrently for the same mountpoint?

Reply:
I guess that concurrent map access race showed that we do.

Reply:
I don't think so (hopefully). I think we don't need to lock here

Reply:
ok then

Reply:
> I guess that concurrent map access race showed that we do.

That was a concurrent access to `mountpoints` map, not to one `mountpoints` element